api/v1alpha1/watcher_types.go

```diff
@@ -52,18 +52,8 @@ const (
 
 // GatewayConfig is used to select an Istio Gateway object in the cluster.
 type GatewayConfig struct {
-	// NamespacedName takes precedence over LabelSelector if configured. Format to use: "namespaceName/gatewayName"
-	// TODO: Add validation: <name><slash><namespace>
-	// +kubebuilder:validation:Optional
-	// +kubebuilder:validation:MinLength=3
-	// +kubebuilder:validation:MaxLength=127
-	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?[/][a-z]([-a-z0-9]*[a-z0-9])?$`
-	NamespacedName string `json:"namespacedName,omitempty"`
-
 	// LabelSelector allows to select the Gateway using label selectors as defined in the K8s LIST API.
-	// Ignored if NamespacedName is set.
-	// +kubebuilder:validation:Optional
-	LabelSelector *metav1.LabelSelector `json:"selector,omitempty"`
+	LabelSelector *metav1.LabelSelector `json:"selector"`
```

Comment:
if this is  no longer optional, I do not think we need a pointer here.

Reply:
yes, also changed `WatcherSpec.Gateway` to non optional, I think now for each Watcher, it must have a related gateway.